pylabnet/scripts/nanopositioners/mcs2_control.py

```diff
@@ -0,0 +1,295 @@
+import numpy as np
+import time
+
+from pylabnet.gui.pyqt.gui_handler import GUIHandler
+from pylabnet.utils.logging.logger import LogHandler
+from pylabnet.utils.helper_methods import unpack_launcher, generate_widgets
+from pylabnet.network.client_server import smaract_mcs2
+
+
+class Controller:
+    """ A script class for controlling MCS2 positioners + interfacing with GUI"""
+
+    NUM_CHANNELS = 9
+    WIDGET_DICT = dict(
+        step_left=NUM_CHANNELS, step_right=NUM_CHANNELS, walk_left=NUM_CHANNELS,
+        walk_right=NUM_CHANNELS, n_steps=NUM_CHANNELS, is_moving=NUM_CHANNELS,
+        amplitude=NUM_CHANNELS, frequency=NUM_CHANNELS, velocity=NUM_CHANNELS, voltage=NUM_CHANNELS
+    )
+    DC_TOLERANCE = 0.1
+
+    def __init__(self, nanopos_client: smaract_mcs2.Client , gui_client, log_client=None):
+        """ Instantiates the controller
+
+        :param nanopos_client: (pylabnet.network.client_server.smaract_mcs2.Client)
+        :param gui_client: (pylabnet.network.client_server.external_gui.Client)
+        :param log_client: (pylabnet.utils.logging.logger.LogClient)
+        """
+
+        self.pos = nanopos_client
+        self.log = LogHandler(logger=log_client)
+        self.gui = GUIHandler(gui_client=gui_client, logger_client=self.log)
+
+        # Unpack all widgets
+        (self.step_left, self.step_right, self.walk_left, self.walk_right,
+         self.n_steps, self.is_moving, self.amplitude, self.frequency,
+         self.velocity, self.voltage) = generate_widgets(self.WIDGET_DICT)
+
+        # Additional attributes
+        self.prev_amplitude = [50]*self.NUM_CHANNELS
```

Comment:
these are the desired default parameters. in the future, we could have these as an input to the script, if that would be useful. but for now they are set by hand here. it's ok since you can easily modify these parameters from the GUI